trip_app/trips/views.py

```diff
@@ -43,3 +53,30 @@ def get_queryset(self):
                 order_by(F('dist1') + F('dist2'))
 
         return queryset
+
+    @action(detail=True, methods=['post'])
+    def reserve(self, request, pk=None):
+        """ Reserve a trip by sending a POST to api/trip/<trip_id>/reserve/.
+            If the man_approve field of the trip model is True the trip request will be created,
+            which means that user requested a trip should be approved by the driver.
+            If man_approve is False user will be added to the passengers list if there are free seats """
+        trip = self.get_object()
+        user = request.user
+        if user == trip.driver:
+            return Response(data="Driver can't be a passenger at the same time", status=status.HTTP_400_BAD_REQUEST)
+        if trip.free_seats:
+            if trip.man_approve:
+                trip_request = TripRequest.objects.create(trip=trip, user=user)
+                serializer = TripRequestSerializer(trip_request)
+                return Response(serializer.data)
+            trip.passengers.add(user)
+            serializer = self.get_serializer(trip)
+            return Response(serializer.data)
```

Comment:
Same here.

Reply:
I'm leaning towards 200 response here because in this case no resource is created but passenger is just added to the trip passengers list, and I return the trip instance in this case.

Reply:
the main item here is the resource (not the underlying models and/or relations). That's why it's up to your general choice of a status for created resources. Keep it 200 everywhere if you prefer so.

Reply:
Thanks for the clarification @andreybulygin. I'm okay with returning either 201/200 responses. But I'm afraid that still got some confusion on this. Let me explain my view on this and try to advocate my point:
So within this particular endpoint I want to reserve a trip. And 2 cases might happen: 
1) if a trip can be auto reserved I just add the user sent the request to the passengers list. And in this case I think that 200 is more relevant, because resource (which is trip in this case) is not created as 201 really assumes, it is updated. And to show that trip resource is updated I add the trip resource in the response. 
2) If a trip assumes an approval from driver, trip request resource is created. And in this case I want to return 201 + trip request entity in the response.
My intuition says me that this is the correct way to do this. Though may be it is correct to stay 201 with POST and 200 with PUT to prevent messing things.

Reply:
REST doesn't have any strict and MUST rules in spec, and thats is veeery big minus of it. So be comfort to use what you think is correct.